src/encoded/search_views.py

```diff
@@ -830,11 +836,20 @@ def rnaget_autocomplete(context, request):
 
 
 @view_config(route_name='rnaget-search', request_method='GET', permission='search')
+@conditional_cache(
+    cache=get_redis_lru_cache(),
+    condition=should_cache_search_results,
+    key=partial(make_key_from_request, 'rnaget-search'),
+)
 def rnaget_search(context, request):
     # Note the order of rendering matters for some fields, e.g. AllResponseField and
     # NotificationResponseField depend on results from BasicSearchWithFacetsResponseField.
     rna_client = request.registry[RNA_CLIENT]
-    fr = FieldedResponse(
+    CachedFieldedResponse = cached_fielded_response_factory(
```

Comment:
Just an idea, not a blocker, but would it make sense to move the caching logic that the decorator is performing inside of CachedFieldResponse as it's the only place being used?

Reply:
Yeah definitely a good idea and probably avoids some of the weird decorator <-> cache binding issues. On the plus side now we have a conditional cache decorator that can be used on ~any view, not just FieldedResponse.